semtypes/src/main/java/io/ballerina/semtype/Conjunction.java

```diff
@@ -0,0 +1,37 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package io.ballerina.semtype;
+
+/**
+ * Represents the Conjunction record type.
+ *
+ * @since 2.0.0
+ */
+public class Conjunction {
+    public Atom atom;
+    public Conjunction next;
+
+    public Conjunction(Atom atom, Conjunction next) {
```

Comment:
Shall we make this private

Reply:
just the constructor or the fields as well? if latter we'll have to create field access methods as well